src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php

```diff
@@ -48,24 +51,21 @@ public function enterNode(Node $node, Environment $env): Node
         }
 
         if ($node instanceof TransDefaultDomainNode) {
-            ++$this->nestingLevel;
-
             if ($node->getNode('expr') instanceof ConstantExpression) {
                 $this->scope->set('domain', $node->getNode('expr'));
 
                 return $node;
             }
 
             if (class_exists(Nodes::class)) {
-                $name = new LocalVariable(null, $node->getTemplateLine());
-                $this->scope->set('domain', $name);
+                $name = new AssignContextVariable(self::INTERNAL_VAR_NAME.(++$this->nameCounter), $node->getTemplateLine());
+                $this->scope->set('domain', new ContextVariable(self::INTERNAL_VAR_NAME.(++$this->nameCounter), $node->getTemplateLine()));
```

Comment:
Incrementing the name counter twice is broken. You won't read the same variable than what you assign.